archivist/archivist.py

```diff
@@ -439,7 +462,15 @@ def count(self, path, *, query=None):
 
         return int(response.headers[HEADERS_TOTAL_COUNT])
 
-    def list(self, path, field, *, page_size=None, query=None, headers=None):
+    def list(
+        self,
+        path: str,
+        field: str,
+        *,
+        page_size: Optional[int] = None,
+        query: Optional[dict] = None,
+        headers: Optional[dict] = None,
+    ):
```

Comment:
I had trouble getting the `Generator` Return type to play well with pylint and the unit test - so In the end I decided to omit it from this PR

Reply:
yes - I would prefer that we keep pylint - is there a later version of pylint that fixes this?